File: zenwave-jpa-example/src/test/java/io/zenwave360/example/infrastructure/jpa/BaseRepositoryIntegrationTest.java
package io.zenwave360.example.infrastructure.jpa;

import io.zenwave360.example.config.DockerComposeInitializer;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
@ActiveProfiles("test")
@DockerComposeInitializer.EnableDockerCompose
@org.springframework.transaction.annotation.Transactional
public abstract class BaseRepositoryIntegrationTest {

}

File: examples/kustomer-address-jpa/src/test/kotlin/io/zenwave360/examples/kotlin/repository/jpa/BaseRepositoryIntegrationTest.kt
package io.zenwave360.examples.kotlin.repository.jpa

import io.zenwave360.examples.kotlin.config.DockerComposeInitializer
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
@ActiveProfiles("test")
@DockerComposeInitializer.EnableDockerCompose
@org.springframework.transaction.annotation.Transactional
abstract class BaseRepositoryIntegrationTest